processing/src/main/java/org/apache/druid/segment/join/filter/JoinFilterAnalyzer.java

```diff
@@ -265,6 +286,10 @@ public static JoinFilterPreAnalysis computeJoinFilterPreAnalysis(
                   (rhsCol) -> Optional.of(new ArrayList<>())
               );
           perColumnCorrelations.get().add(correlationForPrefix.getValue());
+          if (rhsRewriteCandidate.isJoinColumn()) {
+            // we don't need to determine correlated values if the filter is on the join column
```

Comment:
Shouldn't we need to do this?

What if the filter could not be re-written directly (eg. an expression filter, or there were multiple columns in a filter)

Reply:
Hm, I think that `isJoinColumn ` property is more correctly called `isDirectRewrite`, I renamed that.

It's not necessary since if the direct rewrite couldn't happen, then it should fall through to the blocks in `computeJoinFilterPreAnalysis` and  `analyzeJoinFilterClause` which handle indirect rewrites, and it would be handled in that way.

There was a bug that would cause this to not fall through because of missing `supportsRequiredColumnRewrite` checks, I've fixed that and added a filter to `JoinFilterAnalyzerTest.test_filterPushDown_factToRegionToCountryLeftFilterOnRHSJoinConditionColumnsHelper` which overrides `supportsRequiredColumnRewrite` on a SelectorFilter to demonstrate this.